File: 3351-maximize-happiness-of-selected-children/solution.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        
        total_happiness = 0
        for i in range(k):
            current_val = happiness[i] - i
            if current_val <= 0:
                break
                
            total_happiness += current_val
            
        return total_happiness

File: 3351-maximize-happiness-of-selected-children/solution.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& a, int k) {
        long long res = 0, i;
        sort(a.begin(), a.end());
        reverse(a.begin(), a.end());
        for (i = 0; i < k; ++i) res += max(0LL, a[i] - i);
        return res;
    }
};
